occurrence-download/src/main/java/org/gbif/occurrence/download/file/common/DownloadFileUtils.java

```diff
@@ -45,6 +52,27 @@ public static void appendAndDelete(String inputFileName, OutputStream outputFile
       inputFile.delete();
     }
   }
+  
+  /**
+   * reads species count from species list count table path.
+   * @param nameNode namenode of hdfs.
+   * @param path species count table path.
+   * @return species count.
+   * @throws IOException
+   */
+  public static long readSpeciesCount(String nameNode,String path) throws IOException {
+    FileSystem hdfs = DownloadFileUtils.getHdfs(nameNode);
+    for (FileStatus fs : hdfs.listStatus(new Path(path))) {
```

Comment:
does Hive produce more than 1 file? if that's the case we can assume that there's always only 1 file:
Arrays.stream(getHdfs(nameNode).listStatus(new Path(path)))
      .filter(FileStatus::isFile).findFirst()
      .map(fs ->{
        try (BufferedReader countReader = new BufferedReader(new InputStreamReader(hdfs.open(fs.getPath()),
          StandardCharsets.UTF_8))) {
          return Long.parseLong(countReader.readLine());
        }
        }).orElse(0L);